src/browser-sandbox/pruneNonCriticalCss.js

```diff
@@ -131,51 +131,121 @@ export default function pruneNonCriticalCss ({
     return aboveFold
   }
 
-  function isCssRuleCritical (rule) {
-    if (rule.type === 'rule') {
-      // check what, if any selectors are found above fold
-      rule.selectors = rule.selectors.filter(isSelectorCritical)
-      return rule.selectors.length > 0
+  async function isCssSelectorRuleCritical (rule) {
+    // check what, if any selectors are found above fold
+    let selectors = []
+    if (rule.prelude.type === 'SelectorList') {
+      selectors = await window.translateSelectorListRuleToStrings(rule)
+    }
+    if (selectors.length === 0) {
+      return false
+    }
+
+    // filter out the ones that are not critical
```

Comment:
This part may be simpler:
```js
const criticalSelectors = selectors.filter(isSelectorCritical);
const keepRule = criticalSelectors.length > 0;

if (keepRule) {
    rule.prelude.children = criticalSelectors;
}

return keepRule;
```

Reply:
This is my array of selector strings, _not_ the `rule.prelude.children` that are part of the `css-tree` AST - that's why my code is more complicated.

In this PR in general I have mostly kept the existing code structure, but I'm aware it's not the best fit for css-tree. It would be good to explore refactoring to stay closer to css-tree’s api's to see what effect that can have both on performance and on code quality - but that will be for follow up PR's.

Reply:
@lahmatiy The insight I never gave you before while talking about this is that the selector strings are eventually uses to query the browser DOM (`document.querySelector`), that is why I need them in this format.

Reply:
Yep, I see that `selectors` is an array of strings. Since you use `rule.prelude.children.filter()` I suppose children is an array too. So suggested code should works (however I'm not sure about it)

Reply:
The code could be cleaner and better commented - I will look into that going forward - here I just want to get this PR into a mergable state as it dragged a bit over time vs what I had originally wanted. :)

`rule.prelude.children` is indeed an array, but not of strings, it’s the parsed csstree ast values. It’s simply an array because I call `csstree.toPlainObject` early on.

Hence here we have one array of csstree prelude children, and one of css selector strings, hence why the code is as it is. Surely can be improved.